qa/rpc-tests/excessive.py

```diff
@@ -12,17 +12,231 @@
 from test_framework.util import assert_equal
 from test_framework.util import *
+from test_framework.blocktools import *
+import test_framework.script as script
 import pdb
+import sys
+import logging
+logging.basicConfig(format='%(asctime)s.%(levelname)s: %(message)s', level=logging.INFO)
 
 class ExcessiveBlockTest (BitcoinTestFramework):
+    def __init__(self,extended=False):
+      self.extended = extended
+      BitcoinTestFramework.__init__(self)
 
-    def run_test (self):
-        BitcoinTestFramework.run_test (self)
+    def setup_network(self, split=False):
+        self.nodes = start_nodes(4, self.options.tmpdir,timewait=60*10)
+        interconnect_nodes(self.nodes)
+        self.is_network_split=False
+        self.sync_all()
+
+        if 0:  # Use to create wallet with lots of addresses
+          TEST_SIZE=10000
+          print "Creating addresses..."
+          addrs = [ self.nodes[0].getnewaddress() for _ in range(TEST_SIZE+1)]
+          with open("walletAddrs.json","w") as f: 
+            f.write(str(addrs))
+            pdb.set_trace()
+
+    def run_test(self):
+      BitcoinTestFramework.run_test (self)
+
+      # exact point in chain is not required
+      #tips = self.nodes[0].getchaintips ()
+      #assert_equal (len (tips), 1)
+      #assert_equal (tips[0]['branchlen'], 0)
+      #assert_equal (tips[0]['height'], 200)
+      #assert_equal (tips[0]['status'], 'active')
+
+      self.testExcessiveBlock()
+      self.testExcessiveTx()
+
+    def testExcessiveTx(self):
+      TEST_SIZE=20
+      logging.info("Test excessive transactions")
+      if 1:
         tips = self.nodes[0].getchaintips ()
-        assert_equal (len (tips), 1)
-        assert_equal (tips[0]['branchlen'], 0)
-        assert_equal (tips[0]['height'], 200)
-        assert_equal (tips[0]['status'], 'active')
+        #assert_equal (len (tips), 1)
+        #assert_equal (tips[0]['branchlen'], 0)
+        #assert_equal (tips[0]['height'], 200)
+        #assert_equal (tips[0]['status'], 'active')
+
+        self.nodes[0].set("net.excessiveAcceptDepth=0")
+        self.nodes[1].set("net.excessiveAcceptDepth=1")
+        self.nodes[2].set("net.excessiveAcceptDepth=2")
+        self.nodes[3].set("net.excessiveAcceptDepth=3")
+
+        self.nodes[0].set("net.excessiveBlock=2000000")
+        self.nodes[1].set("net.excessiveBlock=2000000")
+        self.nodes[2].set("net.excessiveBlock=2000000")
+        self.nodes[3].set("net.excessiveBlock=2000000")
+
+        logging.info("Cleaning up node state")
+        for n in self.nodes:
+          n.generate(10)
+  	  self.sync_blocks()
+        
+        self.sync_all()
+        # verify mempool is cleaned up on all nodes
+	mbefore = [ (lambda y: (y["size"],y["bytes"]))(x.getmempoolinfo()) for x in self.nodes]
+	assert_equal(mbefore,[(0, 0)]*4)
+
+
+        if 1:
+  	  logging.info("Creating addresses...")
+	  addrs = [ self.nodes[0].getnewaddress() for _ in range(TEST_SIZE+1)]
+        else:
```

Comment:
I added comment.  It allows pre-generation of wallet addresses, but requires some manual steps so a parameter would just be confusing.  The code in "if 0:" is basically notes left to the next engineer describing shortcuts or debugging techniques that may be useful.